pkg/registry/registry.go

```diff
@@ -125,3 +125,24 @@ func IsTransientError(err error) bool {
 		return false
 	}
 }
+
+// IsLoggedIntoRegistryButDontHavePermissions returns true when the error is because the user is logged into the registry but doesn't have permissions to push the image
+func IsLoggedIntoRegistryButDontHavePermissions(err error) bool {
+	return strings.Contains(err.Error(), "insufficient_scope: authorization failed")
+}
+
+// IsNotLoggedIntoRegistry returns true when the error is because the user is not logged into the registry
+func IsNotLoggedIntoRegistry(err error) bool {
+	return strings.Contains(err.Error(), "failed to authorize: failed to fetch anonymous token")
+}
+
+// SplitRegistryAndImage returns image tag and the registry to push the image
+func SplitRegistryAndImage(tag string) (string, string) {
```

Comment:
There is a `GetRepoNameAndTag` function that we might reuse here with a little refactor